Signal/src/ViewControllers/HomeView/HomeViewController.m

```diff
@@ -296,16 +299,29 @@ - (void)viewDidLoad
     }
     
     // Search
-    
+
+    UISearchBar *searchBar = [UISearchBar new];
+    _searchBar = searchBar;
+    searchBar.searchBarStyle = UISearchBarStyleMinimal;
+    searchBar.placeholder = NSLocalizedString(@"HOME_VIEW_CONVERSATION_SEARCHBAR_PLACEHOLDER",
+        @"Placeholder text for search bar which filters conversations.");
+    searchBar.backgroundColor = [UIColor whiteColor];
+    searchBar.delegate = self;
+    [searchBar sizeToFit];
+
     // Setting tableHeader calls numberOfSections, which must happen after updateMappings has been called at least once.
+    OWSAssert(self.tableView.tableHeaderView == nil);
+    self.tableView.tableHeaderView = self.searchBar;
+
     ConversationSearchViewController *searchResultsController = [ConversationSearchViewController new];
     self.searchResultsController = searchResultsController;
-    UISearchController *searchController = [[UISearchController alloc] initWithSearchResultsController:searchResultsController];
-    self.searchController = searchController;
-    searchController.searchResultsUpdater = self;
-    self.tableView.tableHeaderView = self.searchController.searchBar;
-    self.definesPresentationContext = YES;
-        
+    [self addChildViewController:searchResultsController];
+    [self.view addSubview:searchResultsController.view];
+    [searchResultsController.view autoPinWidthToSuperview];
+    [searchResultsController.view autoPinEdge:ALEdgeTop toEdge:ALEdgeBottom ofView:searchBar];
```

Comment:
I declined to do anything fancier for now. It should only be possible to interact with the search bar when the normal home view content is already scrolled to the top of the screen. If necessary we could ensure that.